src/config.ts

```diff
@@ -73,6 +61,31 @@ export const loadConfigurationFromLocalEnv = <T>(configObj: T): T => {
   return configObj;
 };
 
+/**
+ * Updates env variable(s) from process.env
+ * Supports multiple per value
+ * @param value Value from config file to be replaced with env vars
+ */
+export const updateVariableWithLocalEnv = (value: string): string => {
+  const regexp = /\${env:([a-zA-Z_$][a-zA-Z_$0-9]+)}/g;
+  const original = value;
+  let matches = regexp.exec(value as string);
+  while (matches) {
+    if (matches.length > 1) {
+      if (process.env[matches[1]]) {
+        value = value.replace(matches[0], process.env[matches[1]] as string);
+      } else {
+        logger.error(`Env variable needs to be defined for ${matches[1]}`);
+        throw Error(
+          `Environment variable needs to be defined for ${matches[1]} since it's referenced in the config file.`
+        );
+      }
+      matches = regexp.exec(original as string);
```

Comment:
[change request]
we may have a chance to get into an infinite loop. I thought this liner should be
```
matches = regexp.exec(value);
```

Reply:
`value` is being updated inside the loop. Based on docs if you want to iterate through all matches with regex, you need to call the method on the original string over and over. It's not intuitive at all, agreed. ([reference](https://stackoverflow.com/a/432503))